bigquery_etl/dependency.py

```diff
@@ -0,0 +1,86 @@
+"""Build and use query dependency graphs."""
+
+import sys
+from pathlib import Path
+from subprocess import CalledProcessError
+from typing import Tuple
+
+import click
+import jnius_config
+
+# this has to run before jnius can be imported
+for path in (Path(__file__).parent.parent / "target" / "dependency").glob("*.jar"):
+    jnius_config.add_classpath(path.resolve().as_posix())
+
+
+def extract_table_references(sql: str):
+    """Return a list of tables referenced in the given SQL."""
+    # import jnius here so this module can be imported safely without java installed
+    import jnius  # noqa: E402
+
+    try:
+        Analyzer = jnius.autoclass("com.google.zetasql.Analyzer")
+        AnalyzerOptions = jnius.autoclass("com.google.zetasql.AnalyzerOptions")
+    except jnius.JavaException:
+        # replace jnius.JavaException because it's not available outside this function
+        raise ImportError(
+            "failed to import java class via jni, please download java dependencies "
+            "with: mvn dependency:copy-dependencies"
+        )
```

Comment:
We should probably also update the [test docs](https://github.com/mozilla/bigquery-etl/blob/master/tests/README.md) and add `mvn dependency:copy-dependencies` there.